src/hiphip/type_impl.clj

```diff
@@ -1,6 +1,8 @@
 ;; As a hack to avoid writing macro-macros, this file defines the
 ;; per-type macros, and is loaded in each type's namespace.
 
+(def warn-on-reflection *warn-on-reflection*)
```

Comment:
maybe `(def ^:private old-warn-on-reflection *warn-on-reflection*)`? (private, and slightly more descriptive)

Reply:
Sounds good... the PR is updated.